app/src/test/java/com/onewelcome/showcaseapp/viewmodel/MobileAuthenticationEnrollmentViewModelTest.kt

```diff
@@ -0,0 +1,249 @@
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.HiltTestApplication
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+
+@HiltAndroidTest
+@Config(application = HiltTestApplication::class)
+@RunWith(RobolectricTestRunner::class)
+class MobileAuthenticationEnrollmentViewModelTest {
+
+  @get:Rule
+  val hiltRule = HiltAndroidRule(this)
+
+  @Inject
+  lateinit var isSdkInitializedUseCase: IsSdkInitializedUseCase
+
+  @Inject
+  lateinit var getAuthenticatedUserProfileUseCase: GetAuthenticatedUserProfileUseCase
+
+  @Inject
+  lateinit var enrollForMobileAuthenticationUseCase: EnrollForMobileAuthenticationUseCase
+
+  @Inject
+  lateinit var omiSdkFacade: OmiSdkFacade
+
+  @Inject
+  lateinit var oneginiClientMock: OneginiClient
+
+  private val userClientMock = mock<UserClient>()
+  private val oneginiMobileAuthEnrollmentError = mock<OneginiMobileAuthEnrollmentError>()
+
+  private lateinit var viewModel: MobileAuthenticationEnrollmentViewModel
+
+  @Before
+  fun setup() {
+    hiltRule.inject()
+    whenever(oneginiClientMock.getUserClient()).thenReturn(userClientMock)
+  }
+
+  @Test
+  fun `Given SDK is not initialized, When view model is initialized, Then default state should be returned`() {
+    initializeViewModel()
+
+    assertThat(viewModel.uiState).isEqualTo(DEFAULT_STATE)
+  }
+
+  @Test
+  fun `Given SDK is initialized and no user profile in authenticated, When view model is initialized, Then state should be updated`() {
+    givenSdkInitializedAndNoUserAuthenticated()
+
+    assertThat(viewModel.uiState).isEqualTo(
+      DEFAULT_STATE.copy(
+        isSdkInitialized = true
+      )
+    )
+  }
+
+  @Test
+  fun `Given SDK is initialized and user profile is authenticated, When view model is initialized, Then state should be updated`() {
+    givenSdkInitializedAndUserAuthenticated()
+
+    assertThat(viewModel.uiState).isEqualTo(
+      DEFAULT_STATE.copy(
+        isSdkInitialized = true,
+        authenticatedUserProfile = AUTHENTICATED_USER_PROFILE
+      )
+    )
+  }
+
+  @Test
+  fun `Given SDK is not initialized, When EnrollForMobileAuthentication event is sent, Then error result should be returned`() {
+    val expectedException = IllegalStateException("Onegini SDK instance not yet initialized")
+    initializeViewModel()
+
+    viewModel.onEvent(UiEvent.EnrollForMobileAuthentication)
+
+    assertThat(viewModel.uiState)
+      .usingRecursiveComparison()
+      .withEqualsForType(ThrowableEquals(), Throwable::class.java)
+      .isEqualTo(DEFAULT_STATE.copy(enrollmentResult = Err(expectedException)))
```

Comment:
I found that this way we can compare the Results with errors. Adding custom "equals" for Throwable type allows to compare content of different instances. Let me know what do you think. I can adjust other tests to this as well.

Reply:
I'm fine, we can go with this one if you like it more 👍